### PR TITLE
Move NetworkIdentity into protobuf

### PR DESCRIPTION
**Pull request by juniuszhou**

### CODE REVIEW

File: x/oracle/types/networkIdentity.go [hunk 1]
```diff
@@ -19,34 +14,31 @@ func NewNetworkIdentity(networkDescriptor NetworkDescriptor) NetworkIdentity {
...
+	return append(ConsensusNeededPrefix, bytebuf...)
 }
 
 // GetFromPrefix return a NetworkIdentity from prefix
-func GetFromPrefix(key []byte) (NetworkIdentity, error) {
-	if len(key) == 5 {
-		var data NetworkDescriptor
-		bytebuff := bytes.NewBuffer(key[1:])
-		err := binary.Read(bytebuff, binary.BigEndian, &data)
+func GetFromPrefix(cdc codec.BinaryCodec, prefix []byte, key []byte) (NetworkIdentity, error) {
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/oracle/types/networkIdentity.go:35]: the method is unreasonable, will refactor it and add doc.

File: x/ethbridge/keeper/globalNonce.go [hunk 2]
```diff
@@ -87,8 +85,8 @@ func (k Keeper) SetGlobalSequenceToBlockNumber(
 
 // getGlobalSequenceToBlockNumberPrefix
 func (k Keeper) getGlobalSequenceToBlockNumberPrefix(ctx sdk.Context, networkDescriptor oracletypes.NetworkDescriptor, globalNonce uint64) []byte {
-	bs := make([]byte, 4)
-	binary.LittleEndian.PutUint32(bs, uint32(networkDescriptor))
+	networkIdentity := oracletypes.NewNetworkIdentity(networkDescriptor)
+	bs := k.cdc.MustMarshal(&networkIdentity)
 	tmpKey := append(types.GlobalNonceToBlockNumberPrefix, bs[:]...)
 
 	bs = make([]byte, 8)
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/ethbridge/keeper/globalNonce.go:92]: yes, it can make the encode/decode elegant.

[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/ethbridge/keeper/globalNonce.go]: Yes, the binary encode/decode with bigEndian and littleEndian is annoying. I will remove all of them, and use the protobuf instead.

File: x/oracle/types/networkIdentity.go [hunk 3]
```diff
@@ -19,34 +14,31 @@ func NewNetworkIdentity(networkDescriptor NetworkDescriptor) NetworkIdentity {
...
+	if len(prefix) == 1 && len(key) >= 1 && prefix[0] == key[0] {
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/oracle/types/networkIdentity.go]: will refactor it.

File: x/oracle/types/networkIdentity.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/oracle/types/networkIdentity.go]: implementation is not reasonable, will rewrite it.

**Comment by juniuszhou** [comment]
> Would like a little more comments/documentation, otherwise, it's a good change.

binary encode/decode is replaced by proto buf.

**Review by banshee — APPROVED** [review]
LGTM, assuming the earlier comments have been addressed (and it looks like they have).

**Pull request merged**